### DISCUSSION TITLE
Help with folder naming for downloads

### DISCUSSION THREAD

**Discussion by alirz1** [Flexget/Flexget#3657] [General]
Hi, Would like some help here.
I set my tinymedamanager program to sort my tv shows as follows:

Folder: Show title - (year)
Season: S(season#) e.g S1
Episode: S(season#)E(episode#) - episode title        e.g S01E01 - episode name

My flexget is set to Download and name folders/files as:
        path: /volume1/data/Shows/{{series_name}}/S{{ series_season }}
        content_filename: "{{ series_name|replace(' ','.') }}.{{ series_id }}"

Can i make flexget download in the same naming format at least for the folder name i.e Show Title - (year) ? Can flexget determine year with some lookup at download time? THanks

**Comment by BrutuZ** [comment]
Depending on the lookup you're using it might provide a field with the year.
- https://flexget.com/Plugins/thetvdb_lookup has `{{tvdb_first_air_date|formatdate('%Y')}}`
- https://flexget.com/Plugins/tmdb_lookup has `{{tmdb_year}}`
- https://flexget.com/Plugins/trakt_lookup has `{{trakt_series_year}}`
- https://flexget.com/Plugins/tvmaze_lookup has `{{tvmaze_series_year}}`

**Reply by alirz1** [reply]
Awesome, the program is using TMDB for lookup so ill try {{tmdb_year}} Thanks.

**Comment by alirz1** [comment]
one more thing. This YAML is very confusing to me. Where would i put "tmdb_lookup: yes" in the config.yml. Below is my config.
Thank you.

templates:
  global:
    notify:
      entries:
        via:
           - pushbullet:
               api_key: xxxxxxxxxxxxx
               device:  xxxxxxxxxxxxxx
               
tasks:
  ShowRss:
    rss: http://xxxxxxxxxxxxxxxxxxxxxxxxxxx
    all_series: yes
    transmission:
      host: x.x.x.x
      port: xxxx
      username: xxx
      password: xxxxx
      magnetization_timeout: 45
      main_file_only: yes
      path: /volume1/data/Shows/{{series_name} - {{tmdb_year}}/S{{ series_season }}
      content_filename: "{{ series_name|replace(' ','.') }}.{{ series_id }}"

**Reply by BrutuZ** [reply]
```yaml
...
tasks:
  ShowRss:
    rss: https://xxxxxxxxxx
    tmdb_lookup: yes
    all_series: yes
    ...
```

**Reply by alirz1** [reply]
Thank you again.

**Comment by alirz1** [comment]
So i set download path to: path: /volume1/data/Shows/{{series_name}-{{ tmdb_year }}/S{{ series_season }}
But i get error in log: Error setting path for SHOW NAME XXXXX 720p AMZN WEBRip DDP5 1 x264 NTb: Error in template syntax: unexpected '}'

Is the above path incorrect to add the tmdb year to the series folder name? e.g My Show - (2022)

**Reply by BrutuZ** [reply]
You're missing a closing curly brace `}` there

**Reply by alirz1** [reply]
Thanks. Didnt notice that. The syntax all seems to work now However flexget says TMDB lookup failed for the show. Perhaps its not matching anything yet. Ill keep playing with it. Thanks for all the help

**Comment by alirz1** [comment]
For some reason tvdb lookup seems to be failing for everything:

"Doom Patrol S04E06 720p WEB h264 TRUFFLE" torrent added to transmission
2023-01-14 11:27:30
WARNING
tmdb_lookup
ShowRss
TMDB lookup failed for Doom Patrol S04E06 720p WEB h264 TRUFFLE
2023-01-14 11:27:30
VERBOSE
api_tmdb
ShowRss
Searching from TMDb <title=Doom Patrol S04E06, year=None, tmdb_id=None, imdb_id=None>
2023-01-14 11:27:02
INFO
transmission
ShowRss
"Echo 3 S01E09 720p WEB h264 TRUFFLE" torrent added to transmission
2023-01-14 11:27:01
WARNING
tmdb_lookup
ShowRss
TMDB lookup failed for Echo 3 S01E09 720p WEB h264 TRUFFLE
2023-01-14 11:27:01
VERBOSE
api_tmdb
ShowRss
Searching from TMDb <title=Echo 3 S01E09, year=None, tmdb_id=None, imdb_id=None>
2023-01-14 11:26:47
INFO
transmission
ShowRss
"Tulsa King S01E09 720p WEB h264 TRUFFLE" torrent added to transmission
2023-01-14 11:26:47
WARNING
tmdb_lookup
ShowRss
TMDB lookup failed for Tulsa King S01E09 720p WEB h264 TRUFFLE
2023-01-14 11:26:47
VERBOSE
api_tmdb
ShowRss
Searching from TMDb <title=Tulsa King S01E09, year=None, tmdb_id=None, imdb_id=None>`
